### DISCUSSION TITLE
PyPi Pyodide Runtime Package

### DISCUSSION THREAD

**Discussion by johnwason** [pyodide/pyodide#5001] [General]
I develop the PyRI Open Source Teach Pendant for robots, which uses Pyodide for the user interface in a browser. The backend is written in Python. This project will often be used in an air-gapped environment so the packaging system needs to be carefully designed to have all the files stored in packages that can be transferred easily on a storage device. My solution has been to store everything in wheel files.

Unfortunately the full Pyodide runtime with all the available packages is too large to upload to PyPi. My solution has been to use a supplemental pip index feed but this is not a great user experience.

I was wondering if the Pyodide project could create a wheel containing the full runtime and request an exemption from PyPi?

**Comment by ryanking13** [comment]
We deploy the full Pyodide distribution in the [GitHub Releases](https://github.com/pyodide/pyodide/releases). Why would you want it distributed as a Python wheel?

**Reply by johnwason** [reply]
The software is distributed using wheels. The software runs a small server locally that serves the Pyodide files. Because the software is going to be installed on devices without any internet connection, all the files have to be self contained. Having everything stored in wheels makes that much easier.

**Reply by ryanking13** [reply]
Sorry, but I don't think putting a full Pyodide distribution into the wheel is a good idea, as Pyodide is not a Python package, and I want to gradually abandon the idea of "distribution", as we have more and more packages that are built out-of-tree.

However, nothing stops you from creating a wheel from the Pyodide distribution, so if you really need to distribute things in the wheel format. You can make a thin wrapper package that wraps Pyodide distribution in to the wheel
